src/TSID/include/BipedalLocomotion/TSID/FeasibleContactWrenchTask.h

```diff
@@ -0,0 +1,132 @@
+
+    std::shared_ptr<iDynTree::KinDynComputations> m_kinDyn; /**< Pointer to a KinDynComputations
+                                                               object */
+public:
+
+    /**
+     * Initialize the task.
+     * @param handler pointer to the parameter handler.
+     * @note The following parameters are required:
+     * |         Parameter Name        |       Type       |                                  Description                                  | Mandatory |
+     * |:-----------------------------:|:----------------:|:-----------------------------------------------------------------------------:|:---------:|
+     * |        `frame_name`           |      `string`    |                  Name of the frame associated to the contact                  |    Yes    |
+     * |       `variable_name`         |      `string`    | Name of the variable contained in `VariablesHandler` describing the contact   |    Yes    |
+     * |       `number_of_slices`      |       `int`      |                     Number of slices used to split 90 deg.                    |    Yes    |
+     * | `static_friction_coefficient` |      `double`    |                          Static friction coefficient.                         |    Yes    |
+     * |        `foot_limits_x`        | `vector<double>` |     x coordinate of the foot limits w.r.t the frame attached to the surface   |    Yes    |
+     * |        `foot_limits_y`        | `vector<double>` |     y coordinate of the foot limits w.r.t the frame attached to the surface   |    Yes    |
+     * @return true in case of success/false otherwise.
+     */
+    bool initialize(std::weak_ptr<const ParametersHandler::IParametersHandler> paramHandler);
+
+    /**
+     * Set the kinDynComputations object.
+     * @param kinDyn pointer to a kinDynComputations object.
+     * @return True in case of success, false otherwise.
+     */
+    bool setKinDyn(std::shared_ptr<iDynTree::KinDynComputations> kinDyn);
+
+    /**
+     * Set the set of variables required by the task. The variables are stored in the
+     * System::VariablesHandler.
+     * @param variablesHandler reference to a variables handler.
+     * @note The handler must contain a variable named as the parameter
+     * `variable_name` stored in the parameter handler. The variable represents the contact wrench
+     * acting of the robot.
+     * @return True in case of success, false otherwise.
+     */
+    bool setVariablesHandler(const System::VariablesHandler& variablesHandler) override;
+
+    /**
+     * Update the content of the element.
+     * @return True in case of success, false otherwise.
+     */
+    bool update() override;
+
+    /**
+     * Enable the task. I.e. a contact wrench different from zero is allowed.
+     */
+    void enable();
+
+    /**
+     * Disable the task. I.e. the contact wrench is forced to be equal to zero.
+     */
+    void disable();
```

Comment:
There is some potential confusion on whether it is the task to be disabled (hence the wrench is not constrained), or if the wrench cannot be used to control the robot. A possibility could be to allow setting directly the maximum normal component. Another possibility could be to have a method like ``setContactState`` with an enum. Otherwise also a different naming could do the trick, although I did not find any good alternatives myself.

Reply:
I like `setContactState`! the enum could be `Active` `Inactive`, what do you think?

Reply:
In theory the enum could be defined in https://github.com/dic-iit/bipedal-locomotion-framework/blob/d81b390683127a451669b0e09ff06da199f9490a/src/Contacts/include/BipedalLocomotion/Contacts/Contact.h#L19

Reply:
That would be nice. On the other hand, we should try to be coherent and use that enum also inside the class. Right now it is a boolean https://github.com/dic-iit/bipedal-locomotion-framework/blob/d81b390683127a451669b0e09ff06da199f9490a/src/Contacts/include/BipedalLocomotion/Contacts/Contact.h#L102
and only in the ``EstimatedContact``.

For the time being, we could simply add the boolean and start using it in separate PRs.

Reply:
In 9b8662b I implemented the function we discussed.